From d8a557f838f7fbd4b99f22c8c0e37ff635e87393 Mon Sep 17 00:00:00 2001
From: Shunqian Zheng <zhengsq@rock-chips.com>
Date: Sat, 14 Jul 2018 10:57:26 +0800
Subject: [PATCH] media: rockchip: isp1: fix wrong fmt of isp_sd .get_fmt

The isp_sd .get_fmt of source pad return format of input by mistake.
It should be the *out_fmt of isp_sd.

Change-Id: I9f592a2906275ef58dc564d68de63aac393c8690
Signed-off-by: Shunqian Zheng <zhengsq@rock-chips.com>
---
 drivers/media/platform/rockchip/isp1/rkisp1.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/platform/rockchip/isp1/rkisp1.c b/drivers/media/platform/rockchip/isp1/rkisp1.c
index 98098ff02db7..88f20b7fcd97 100644
--- a/drivers/media/platform/rockchip/isp1/rkisp1.c
+++ b/drivers/media/platform/rockchip/isp1/rkisp1.c
@@ -691,7 +691,7 @@ static int rkisp1_isp_sd_get_fmt(struct v4l2_subdev *sd,
 		*mf = isp_sd->in_frm;
 	} else if (fmt->pad == RKISP1_ISP_PAD_SOURCE_PATH) {
 		/* format of source pad */
-		*mf = isp_sd->in_frm;
+		mf->code = isp_sd->out_fmt.mbus_code;
 		/* window size of source pad */
 		mf->width = isp_sd->out_crop.width;
 		mf->height = isp_sd->out_crop.height;
-- 
2.35.3

